Sources/ProjectSpec/SpecFile.swift

```diff
@@ -158,7 +158,9 @@ extension Dictionary where Key == String, Value: Any {
     func replaceString(_ template: String, with replacement: String) -> JSONDictionary {
         var replaced: JSONDictionary = self
         for (key, value) in self {
-            replaced[key] = replace(value: value, template, with: replacement)
+            let newKey = key.replacingOccurrences(of: template, with: replacement)
+            replaced.removeValue(forKey: key)
+            replaced[newKey] = replace(value: value, template, with: replacement)
```

Comment:
This change worries me. Here's the scenario I'm looking to support with it:

```yml
schemeTemplates:
  StandardScheme:
    build:
      targets:
       ${moduleName}: build
    test:
      gatherCoverageData: true
      targets:
        - name:${moduleName}Tests
          parallelizable: true
          randomExecutionOrder: true

schemes:
  MyApp:
    templates:
      - StandardScheme
    templateAttributes:
      moduleName: MyApp
```
Since `schemeTemplates.StandardScheme.build.targets` has a key that needs to be configurable, the current implementation of `replace` didn't work. However, I think it will be extremely common when using scheme templates to want to be able to dynamically generate the name of the target you're building and testing against.

Reply:
Should be ok. Though, perhaps we can only remove and set the new key if it has changed (`if newKey != key`)

Reply:
Good idea, I added that.